### PR TITLE
Optimize CPU cycles for persistent sessions

### LINKED ISSUE #31701: Optimize CPU cycles for persistent sessions
### Description

During a performance analysis, some unnecessary CPU cycles have been identified. 

### Discussion

_No response_

### Motivation

* Don't wake up the persistent sessions worker that often, as it will wake up on shutdown anyway due to the interrupting
* Create the changes performer data structure lazily only when needed to save some CPU cycle and GC items 

### Details

I'll create a PR

### PR DESCRIPTION
**Pull request by ahus1**
Closes #31701

### CODE REVIEW

**Comment by ahus1** [comment]
@mhajas - please review when you have the time. Thanks!

File: model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/changes/PersistentSessionsWorker.java [hunk 1]
```diff
@@ -76,7 +76,8 @@ public void run() {
 
         private void process(ArrayBlockingQueue<PersistentUpdate> queue) throws InterruptedException {
             ArrayList<PersistentUpdate> batch = new ArrayList<>();
-            PersistentUpdate polled = queue.poll(100, TimeUnit.MILLISECONDS);
+            // Timeout of 10 seconds is only a backup if interrupting the task didn't work as expected.
```
Comment (mhajas) [inline model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/changes/PersistentSessionsWorker.java:79]: 10 seconds does not correspond to the value below. Also, what do you mean by "if interrupting the task didn't work as expected"?

**Review by mhajas — COMMENTED** [review]
I added one comment otherwise LGTM

**Comment by ahus1** [comment]
@mhajas - thank you, updated the comment to the following, which might reveal more of the context: 

> Timeout is only a backup if interrupting the worker task in the stop() method didn't work as expected because someone else swallowed the interrupted flag.

**Review by mhajas — APPROVED** [review]
Thank you @ahus1

**Pull request merged**